### PR TITLE
Set up ADO build pipeline for ESRP release

### PR DESCRIPTION
**Pull request by kaibocai**
### Issue describing the changes in this PR

1. Set ADO build pipeline for ESRP release engine. 
2. Improve github CI running speed by applying new CI pattern. 

### Pull request checklist

* [X] My changes **do not** require documentation changes
    * [ ] Otherwise: Documentation issue linked to PR
* [X] My changes are added to the `CHANGELOG.md`
* [X] I have added all required tests (Unit tests, E2E tests)

### Additional information

Additional PR information

### CODE REVIEW

File: .github/workflows/build-validation.yml [hunk 1]
```diff
@@ -70,27 +78,32 @@ jobs:
...
+
       - name: Set up JDK 8
         uses: actions/setup-java@v2
         with:
           java-version: '8'
           distribution: 'temurin'
+
+      - name: Setup Gradle
+        uses: gradle/gradle-build-action@v2
+
```
Comment (cgillum) [inline .github/workflows/build-validation.yml:90]: Should this newline be removed?

File: azdevops-pipeline/azure-pipelines.yml [hunk 2]
```diff
@@ -0,0 +1,57 @@
+# Gradle
```
Comment (cgillum) [inline azdevops-pipeline/azure-pipelines.yml:1]: It looks like this is a release pipeline. Can we name this file `azure-pipelines-release.yml` or something similar?

File: azdevops-pipeline/azure-pipelines.yml [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+steps:
+- checkout: self
+  submodules: true
+
+- task: Gradle@3
+  inputs:
+    workingDirectory: ''
+    gradleWrapperFile: 'gradlew'
+    gradleOptions: '-Xmx3072m'
```
Comment (cgillum) [inline azdevops-pipeline/azure-pipelines.yml:19]: What are these options for? Are they something common? If not, can we add a comment explaining what they do?

File: azurefunctions/build.gradle [hunk 4]
```diff
@@ -26,6 +27,11 @@ sourceCompatibility = JavaVersion.VERSION_1_8
 targetCompatibility = JavaVersion.VERSION_1_8
 
 publishing {
+    repositories {
+        maven {
+            url "file://$project.rootDir/repo"
```
Comment (cgillum) [inline azurefunctions/build.gradle:32]: Does this mean we're publishing to the local file system, and then we manually upload to Maven Central?

**Review by cgillum — COMMENTED** [review]
Nice to see the cleanup of the GitHub Actions pipeline yaml.

A few questions.

File: .github/workflows/build-validation.yml (hunk 1 above)
Comment (kaibocai) [reply .github/workflows/build-validation.yml:90]: It's two seperate tasks, the first task install gradle for the entire job, so we don't repeat the installation in each task.  We just run gradle command directly in later tasks.

File: azdevops-pipeline/azure-pipelines.yml (hunk 2 above)
Comment (kaibocai) [reply azdevops-pipeline/azure-pipelines.yml:1]: It's acutally a build pipeline that generated all required artifacts we want to deploy to maven central. Once the artifacts is generated, we use [this release pipeline](https://durabletaskframework.visualstudio.com/Durable%20Task%20Framework%20CI/_release?view=all&definitionId=8) to do the release.

File: azurefunctions/build.gradle (hunk 4 above)
Comment (kaibocai) [reply azurefunctions/build.gradle:32]: We will generate the artifacts to this file system location, and later them will be consumed by [this release pipeline](https://durabletaskframework.visualstudio.com/Durable%20Task%20Framework%20CI/_release?view=all&definitionId=8).

File: azdevops-pipeline/azure-pipelines.yml (hunk 2 above)
Comment (cgillum) [reply azdevops-pipeline/azure-pipelines.yml:1]: Got it - can we rename the file to something like `build-release-artifacts.yml` then?

**Pull request merged**